### PR TITLE
Deflector refactoring

### PR DESCRIPTION
**Pull request by sibirrer**
Added a Deflector() class as a skeleton to develop more diverse deflector classes (such as NFW+Hernquist, clusters etc)

### CODE REVIEW

**Comment by sibirrer** [comment]
@furcelay just fyi, I am updating with this PR the design and made a more flexible structure with a Deflector() class, that can serve as a baseline for more complex deflector models relevant for #134

**Comment by sibirrer** [comment]
@nkhadka21 can you please review this PR. It should not affect your side and it's a step towards having the NFW+Hernquist model implemented - thank you!

File: slsim/Deflectors/deflector.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        :return: e1_mass, e2_mass
+        """
+        e1_mass, e2_mass = float(self._deflector_dict["e1_mass"]), float(
+            self._deflector_dict["e2_mass"]
+        )
+        return e1_mass, e2_mass
+
+    def light_model_lenstronomy(self, band=None):
+        """Returns lens model instance and parameters in lenstronomy conventions.
```
Comment (nkhadka21) [inline slsim/Deflectors/deflector.py:87]: docstring for band?

**Review by nkhadka21 — APPROVED** [review]
Hi @sibirrer, thank you for this PR! It looks good to me. I have a minor comment for a docstring.

**Pull request merged**